### PR TITLE
Add UK bank holidays

### PR DESCRIPTION
**Pull request by SimonBarrettACT**
This adds bank holidays for Wales, England, Scotland and Northern Ireland.

If a bank holiday is on a weekend, a ‘substitute’ weekday becomes a bank holiday, normally the following Monday.

https://www.gov.uk/bank-holidays

### Country codes
Wales: gb-cym
Scotland: gb-sct
Northern Ireland: gb-nir
England: gb-eng

### CODE REVIEW

**Comment by nebarg** [comment]
I don't think the Christmas/Boxing day is correct.

It looks like if Christmas Day is Sunday, Boxing Day stays as Monday and Christmas (substitute) is moved to Tuesday.

An example can be seen for 2022 on https://www.gov.uk/bank-holidays

**Comment by SimonBarrettACT** [comment]
Thanks @nebarg, that is correct. 

If Christmas is on a Friday, Boxing Day is a substitute day on the Monday
If Christmas is on a Saturday, Christmas is a substitute day on Monday and Boxing Day is a substitute day on the Tuesday.

**Comment by nanos** [comment]
Just to say that I believe GB was the correct country code.

The readme references country codes (although it doesn't specify which one). Assuming this means ISO 3166-2 the correct country codes are:

GB-ENG for England
GB-NIR for Northern Ireland
GB-SCT for Scotland
GB-WLS for Wales

(BS 6879 also allows GB-CYM for Wales, but ISO 3166-2 doesn't allow double coding, so this isn't formally part of ISO 3166-2 though most implementations allow both.)

Source: https://www.iso.org/obp/ui/#iso:code:3166:GB

File: src/Countries/Wales.php [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            $christmasDay = $christmasDay->next('monday');
+        }
+
+        if ($christmasDay->isSunday()) {
+            $key .= ' (substitute day)';
+            $christmasDay = $christmasDay->next('tuesday');
+        }
+
+        return [$key => $christmasDay];
+    }
```
Comment (Ikpeba4ll) [inline src/Countries/Wales.php:30]: Great thought for this scenario.

Is it possible you move chrismas, boxing and new year to a generic file so other countries can inherit it.

**Comment by SimonBarrettACT** [comment]
I switched it because NI isn't in Great Britain.

File: src/Countries/NorthernIreland.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        $oneOffHolidays = $this->oneOffHolidays($year);
+
+        return array_merge($regularHolidays, $oneOffHolidays);
+
+    }
+
+    /** @return array<string, CarbonImmutable> */
+    protected function variableHolidays(int $year): array
+    {
+        $easterSunday = CarbonImmutable::createFromTimestamp(easter_date($year))
```
Comment (Nielsvanpach) [inline src/Countries/NorthernIreland.php:77]: Could you use `$this->easter($year)` instead? Available after having your fork in sync & rebasing your branch on our main branch

Comment (Nielsvanpach) [reply src/Countries/NorthernIreland.php:77]: Also for the other usages of easter_date() ofcourse

**Review by Nielsvanpach — COMMENTED** [review]
Hi, thanks for your work on this! Could you also solve the phpstan issues?

Did you verify the generated results in the snapshots?

**Comment by Nielsvanpach** [comment]
I think something went wrong while rebasing: 115 updated files.

File: src/Countries/Wales.php (hunk 1 above)
Comment (SimonBarrettACT) [reply src/Countries/Wales.php:31]: I'm not sure about this as different countries might have different rules when either fall on a weekend.

**Comment by SimonBarrettACT** [comment]
I have opened a new PR request #162  to fix the rebase issue.

**Pull request closed without merging**